storybookjs/storybook
# Preset: Add common preset overrides mechanism

Closes #23914






## What I did

This PR adds a `common-override-presets` file in `core-common` which will get executed early on, before presets from packages like react-webpack5, and with that we can make sure that `framework.options` always exist, for instance (which is what is done in this PR). This can be useful for other scenarios as well in the future.
This will fix the infamous `cannot read properties of undefined reading 'builder'` on all frameworks, including community ones (rspack etc) 

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing


1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Comment out framework.options in main.js:
```ts
const config: StorybookConfig = {
  framework: {
      name:'@storybook/react-vite',
      // options: {} <-- comment this!
  }
}
```
3. See things work


### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
